File: DemoMethods/Basic/MapReduceUsingDocumentQuery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web.Http;
using DemoMethods.Helpers;
using DemoMethods.Indexes;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {
        [HttpGet]
        public object MapReduceUsingDocumentQuery()
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                IList<ProductSales.Result> results = session
                    .Advanced
                    .DocumentQuery<ProductSales.Result, ProductSales>()
                    .ToList();

                return results;
            }
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/basic/MapReduceUsingDocumentQuery.java
package net.ravendb.demo.basic;

import net.ravendb.abstractions.basic.Reference;
import net.ravendb.client.IDocumentSession;
import net.ravendb.client.RavenQueryStatistics;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.indexes.ProductSales;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

import java.util.List;

@Controller
public class MapReduceUsingDocumentQuery {

    @RequestMapping("/Basic/MapReduceUsingDocumentQuery")
    public List<ProductSales.Result> mapReduceUsingDocumentQuery() {
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            Reference<RavenQueryStatistics> statsRef = new Reference<>();

            List<ProductSales.Result> results = session
                    .advanced()
                    .documentQuery(ProductSales.Result.class, ProductSales.class)
                    .statistics(statsRef)
                    .toList();

            return results;
        }
    }
}
